src/main/scala/Logic.scala

```diff
@@ -11,60 +11,70 @@ class Logic {
     var i: Int = 0
     var userScore: Int = 0
     var aiScore: Int = 0
+    val winningScore : Int = 3
+
 
 
     //Users name
     val userName = readLine("Enter your name")
     println(userName)
 
-    //Users choice
-
 
     //Game loop
-    for (x <- 0 to 2) {
-
-      val userInput = readLine("Rock paper scissors...")
-
-      //Users input
-      userInput match {
+    while (userScore < winningScore && aiScore < winningScore) {
 
-        case "rock" => i = 0
-        case "paper" => i = 1
-        case "scissors" => i = 2
-      }
+      val userInput = readLine("Rock, paper, scissors...")
 
+println(userInput)
       //Match user input
-      i match {
-
-        //User chooses rock
-        case 0 =>
+      val aiPick = ai.aiRPS
 
-          if (ai.aiPlay == 0){}
-          else if (ai.aiPlay == 1){aiScore = aiScore + 1}
-          else { userScore = userScore + 1}
-
-        case 1 =>
-          if (ai.aiPlay == 1){}
-          else if(ai.aiPlay == 2) {aiScore = aiScore +1}
-          else {userScore = userScore + 1}
+      userInput match {
 
-        case 2 =>
-          if (ai.aiPlay == 2){}
-          else if( ai.aiPlay == 0) {aiScore = aiScore + 1}
-          else {userScore = userScore}
+        //User chooses rock
+        case "rock" => {
+
+          if (aiPick == "Rock") {aiScore == aiScore && userScore == userScore}
```

Comment:
aiScore == aiScore && userScore == userScore <-- this part as already said is useless, also instead of if elses I'd prefer pattern matching, but not mandatory